src/Compilers/VisualBasic/Test/Semantic/FlowAnalysis/RegionAnalysisTests.vb

```diff
@@ -572,6 +571,7 @@ End Class
             Assert.True(analysis.Succeeded)
             Assert.Equal("j, l", GetSymbolNamesJoined(analysis.DataFlowsIn))
             Assert.Equal("i, k", GetSymbolNamesJoined(analysis.DataFlowsOut))
+            Assert.Equal(Nothing, GetSymbolNamesJoined(analysis.DefinitelyAssignedOnEntry))
```

Comment:
❔ Why is this not `j, l`?

Reply:
@gafter to confirm.  But i think it's because the code unconditionally jumps into the region.  so, since hte start of the region is unreachable, nothing is considered definitely assigned there.

Reply:
In an unreachable region, every variable is definitely assigned.  But they way DefinitelyAssignedOnEntry is implemented, it treats that as no variable assigned.

This may be a reason that "definitely assigned on entry plus always assigned inside" is not the same as definitely assigned on exit.  In short, I think the "definitely assigned on exit" API should be done at the same time.

---
In reply to: [306502473](https://github.com/dotnet/roslyn/pull/37395#discussion_r306502473) [](ancestors = 306502473)

Reply:
> But if it is not needed now, it is harmless to leave it to a future change.

@gafter I'm ok doing this as part of this PR.  But given the amount of work, i woud def want to make sure you think it's appropriate to do (and would be done in a similar fashion to DefinitelyAssignedOnEntry).  Specifically, going and updating all the tests is non-trivial :)   So before i do it, i want ot realllllly make sure i'm doing it right :)